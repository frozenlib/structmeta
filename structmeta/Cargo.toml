[package]
name = "structmeta"
version = "0.1.0"
authors = ["frozenlib"]
license = "MIT OR Apache-2.0"
readme = "../README.md"
repository = "https://github.com/frozenlib/structmeta"
documentation = "https://docs.rs/structmeta/"
keywords = ["derive", "parse", "attribute", "syn", "totokens"]
categories = ["development-tools::procedural-macro-helpers"]
description = "Parse Rust's attribute arguments by defining a struct."
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
structmeta-derive = { version = "=0.1.0", path = "../structmeta-derive" }
proc-macro2 = "1.0.24"
syn = "1.0.65"
quote = "1.0.9"
