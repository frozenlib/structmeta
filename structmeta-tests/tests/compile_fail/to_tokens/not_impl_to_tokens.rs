use structmeta::ToTokens;
#[derive(ToTokens)]
struct TestType {
    x: X,
}

struct X;

fn main() {}
