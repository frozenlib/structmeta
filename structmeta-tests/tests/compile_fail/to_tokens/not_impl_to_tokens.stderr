error[E0277]: the trait bound `X: ToTokens` is not satisfied
 --> tests/compile_fail/to_tokens/not_impl_to_tokens.rs:4:5
  |
4 |     x: X,
  |     ^ the trait `ToTokens` is not implemented for `X`
  |
  = help: the following other types implement trait `ToTokens`:
            &'a T
            &'a mut T
            Box<T>
            Cow<'a, T>
            Option<T>
            Rc<T>
            RepInterp<T>
            String
          and $N others
